test/simulator/moves/encore.js

```diff
@@ -0,0 +1,123 @@
+		hp = battle.p2.active[0].hp;
+
+		// During subsequent turns the normal Focus Punch behavior applies.
+		battle.makeChoices('move focuspunch 1, move teleport', 'move splash, move extremespeed 1');
+		assert.strictEqual(battle.p2.active[0].hp, hp);
+	});
+
+	it('should not affect Shell Trap if the user\'s decision is not changed', function () {
+		battle = common.createBattle({gameType: 'doubles'}, [
+			[
+				{species: "Smeargle", level: 99, ability: 'owntempo', moves: ['shelltrap', 'splash']},
+				{species: "Abra", level: 1, ability: 'innerfocus', moves: ['knockoff', 'teleport']},
+			],
+			[
+				{species: "Smeargle", ability: 'owntempo', item: 'laggingtail', moves: ['encore', 'splash']},
+				{species: "Zigzagoon", ability: 'pickup', item: 'assaultvest', moves: ['quickattack']},
+			],
+		]);
+
+		// If the Shell Trap user is hit the attack is expected to be successful.
+		battle.makeChoices('move shelltrap, move knockoff 1', 'move splash, move quickattack 1');
+		let hp = battle.p2.active[0].hp;
+		assert.notStrictEqual(hp, battle.p2.active[0].maxhp);
+
+		// If a user's previous move was Shell Trap and it is Encored into Shell Trap while attempting to
+		// execute the move, the regular "you must be hit" effect for Shell Trap will be enforced
+		battle.makeChoices('move shelltrap, move teleport', 'move encore 1, move quickattack 1');
+		assert.notStrictEqual(battle.p2.active[0].hp, hp);
+		hp = battle.p2.active[0].hp;
+
+		// During subesquent turns the normal Shell Trap behavior applies.
+		battle.makeChoices('move shelltrap, move teleport', 'move splash, move quickattack 1');
+		assert.notStrictEqual(battle.p2.active[0].hp, hp);
+	});
+
+	it('should make Shell Trap always fail if the user\'s decision is changed', function () {
+		battle = common.createBattle({gameType: 'doubles'}, [
+			[
+				{species: "Smeargle", level: 99, ability: 'owntempo', moves: ['splash', 'shelltrap']},
+				{species: "Abra", level: 1, ability: 'innerfocus', moves: ['knockoff', 'teleport']},
+			],
+			[
+				{species: "Smeargle", ability: 'owntempo', item: 'laggingtail', moves: ['encore', 'splash']},
+				{species: "Zigzagoon", ability: 'pickup', item: 'assaultvest', moves: ['quickattack']},
+			],
+		]);
+
+		// If the Shell Trap user is hit the attack is expected to be successful.
+		battle.makeChoices('move shelltrap, move knockoff 1', 'move splash, move quickattack 1');
+		let hp = battle.p2.active[0].hp;
+		assert.notStrictEqual(hp, battle.p2.active[0].maxhp);
+
+		// Shell Trap which has been encored will never be successful - even if it is hit with contact moves, it will never
+		// attack, and will always say "<Pokemon>'s shell trap didn't work. It doesn't matter in the case of Shell Trap if
+		// the user was hit before or after the Encore; Shell Trap will still always fail.
+		battle.makeChoices('move splash, move teleport', 'move encore 1, move quickattack 1');
+		assert.strictEqual(battle.p2.active[0].hp, hp);
```

Comment:
Without the change to `onTry` for Shell Trap, the only reason this works is because Encore fails against the failed Shell Trap, and Splash predictably deals no damage to the other Smeargle so the test passes. Do we want to just leave it like this until spread move order is fixed?